server/controllers/refer/caseListController.ts

```diff
@@ -0,0 +1,50 @@
+import type { Request, Response, TypedRequestHandler } from 'express'
+import createHttpError from 'http-errors'
+
+import type { ReferralService } from '../../services'
+import { CaseListUtils, PaginationUtils, TypeUtils } from '../../utils'
+
+export default class ReferCaseListController {
+  constructor(private readonly referralService: ReferralService) {}
+
+  show(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      TypeUtils.assertHasUser(req)
+
+      const referralStates: Record<string, Array<string>> = {
```

Comment:
- Should this be `referralStatuses`?
- `Array<ReferralStatus>`?
- I can't remember if you can use a non-primitive type as the first argument(?) in a `Record` type, but I wonder if it's worth adding a UI type for the three referral status groups?